src/Sulu/Bundle/AdminBundle/Resources/js/containers/Datagrid/stores/DatagridStore.js

```diff
@@ -4,17 +4,24 @@ import ResourceRequester from '../../../services/ResourceRequester';
 import metadataStore from './MetadataStore';
 
 export default class DatagridStore {
-    page: observable = observable();
-    locale: observable = observable();
     @observable pageCount: number = 0;
     @observable data: Array<Object> = [];
     @observable selections: Array<string | number> = [];
     @observable loading: boolean = true;
     disposer: () => void;
     resourceKey: string;
+    options: Object;
+    observableOptions: Object;
 
-    constructor(resourceKey: string) {
+    constructor(resourceKey: string, observableOptions: Object = {}, options: Object = {}) {
         this.resourceKey = resourceKey;
+        this.observableOptions = observableOptions;
+
+        if (!this.observableOptions.page) {
+            throw new Error('The DatagridStore expects an observable "page" property inside the "observableOptions".');
```

Comment:
Instead of this I would add a more specific flow type to the `observableOptions` paramter instead of just `Object`.